hazelcast/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java

```diff
@@ -156,17 +159,27 @@ public void test_itemStream_withWindowing() throws Throwable {
     public void test_itemStream_in_expected_range() throws Throwable {
         int itemsPerSecond = 109;
 
+        int windowSize = 1000;
         p.readFrom(TestSources.itemStream(itemsPerSecond))
-                .withNativeTimestamps(0)
-                .window(tumbling(1000))
-                .aggregate(counting())
-                .apply(assertCollectedEventually(10, windowResults -> {
+         .withNativeTimestamps(0)
+         .window(tumbling(windowSize))
+         .aggregate(counting())
+         .apply(assertCollectedEventually(10, windowResults -> {
                     // first window may be incomplete
                     assertTrue("sink list should contain some items", windowResults.size() > 1);
-                    assertTrue("emitted items is more than twice lower then expected itemsPerSecond",
-                            (long) windowResults.get(1).result() > itemsPerSecond / 2);
-                    assertTrue("emitted items is more than twice higher then expected itemsPerSecond",
-                            (long) windowResults.get(1).result() < itemsPerSecond * 2);
+                    WindowResult<Long> windowResult = windowResults.get(1);
+
+                    long pauses = JitterRule.getPausesBetween(windowResult.start(), windowResult.end());
+                    // Ignore if a long pause was recorded
+                    if (pauses < (windowSize / 2)) {
+                        assertThat(windowResult.result())
+                                .isGreaterThan(itemsPerSecond / 2)
+                                .describedAs("emitted items is more than twice lower then expected itemsPerSecond");
+
+                        assertThat(windowResult.result())
+                                .isLessThan(itemsPerSecond * 2)
+                                .describedAs("emitted items is more than twice higher then expected itemsPerSecond");
+                    }
                 }));
 
         expectAssertionsCompleted();
```

Comment:
Is it possible that these assertion won't be run at all? Maybe we could add `AtomicBoolean assertionChecked = new AtomicBoolean();` and then:
 
```suggestion
                        assertionChecked.set(true);
                    }
                }));
        assertThat(assertionChecked).isTrue();
        expectAssertionsCompleted();
```

Reply:
Yes, it is possible and I did that intentionally.

The current situation is: the test failed and I saw a large hiccup reported which explains the failure so it was exceptional -> we can ignore the failure. The change makes this check automated.

If we did what you propose it would fail if such hiccup was detected.

We check only the 2nd window (`windowResults.get(1)`) - the 1st window might be small because the source might start producing just at the end of it.

Another option would be to check all window results and try to find _some_, or at least X which fits the criteria. Wdyt?

Reply:
I thought that we expect that one of retries done by `assertCollectedEventually` will be run without hiccups.